src/queue.ts

```diff
@@ -183,33 +183,46 @@ export class Queue {
   // Job states: new -> started -> finished
 
   async createJob(payload: string): Promise<Job> {
-    this.guardThatLastMessageWasJobFinishedOrNull(this.getLatestMessage())
+    const latestMessage = this.getLatestMessage()
+    this.guardThatLastMessageWasJobFinishedOrNull(latestMessage)
+
+    const newJobId = latestMessage.isNull() ? 0 : latestMessage.jobId() + 1
```

Comment:
@yeraydavidrodriguez we could extract a method `this.nextJobId()` or `this.newJobId()`.

On the other hand, we are going to need to refactor this code when we allow [more than one pending job](https://github.com/Nautilus-Cyberneering/git-queue/issues/30). It's OK, we can refactor it later, but we could search for the latest `NewJobMessage` instead of only the latest message.

Reply:
Yes, you are right. Refactored.